hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/CompactHelpers.java

```diff
@@ -65,6 +66,7 @@ public HoodieCommitMetadata createCompactionMetadata(
       metadata.addWriteStat(stat.getPartitionPath(), stat);
     }
     metadata.addMetadata(org.apache.hudi.common.model.HoodieCommitMetadata.SCHEMA_KEY, schema);
+    metadata.setOperationType(WriteOperationType.COMPACT);
```

Comment:
When I run test in TestWriteMergeOnRead, I found compaction in flink doesn't add operationType to commit meta data.

Reply:
@danny0405 This change has nothing to do with this pr, but the problem exists, should I keep this line？